### PR TITLE
Bump flake8 to v5, flake8-bandit to v4.1, for python 3.8+

### PR DESCRIPTION
**Pull request by sathieu**
# I have made things!

Allow installation with flake8 v5 and flake8-bandit v4.

Keep older version for python 3.7 because of [this](https://github.com/PyCQA/flake8/commit/975a3f45334861ebd8960c00e881443c23654bca#commitcomment-58171414).

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -63,7 +63,7 @@ flake8-bugbear = "^22.9"
 flake8-debugger = "^4.0"
 flake8-isort = "^4.0"
 flake8-eradicate = "^1.0"
-flake8-bandit = ">=2.1,<4"
+flake8-bandit = ">=2.1,<5"
```
Comment (sobolevn) [inline pyproject.toml:66]: Can we bump the lower part?

**Review by sobolevn — CHANGES_REQUESTED** [review]
Please, run `poetry update` as well

File: pyproject.toml (hunk 1 above)
Comment (sathieu) [reply pyproject.toml:66]: @sobolevn Yes. Which lower bound do you want?   `>=3` ?

Comment (sobolevn) [reply pyproject.toml:66]: I prefer to use the exact `^` pin. So is going to be `^4.whatever-it-is`

Comment (sathieu) [reply pyproject.toml]: @sobolevn Done. Is this OK now?

**Comment by sobolevn** [comment]
`poetry update` is still needed :)

**Comment by sobolevn** [comment]
Sorry, we cannot drop 3.7 just yet.
It is widely used, including my own projects.

Is there any other way?

**Review by sobolevn — APPROVED** [review]
This looks much better!

File: tests/test_options/conftest.py [hunk 2]
```diff
@@ -8,6 +9,8 @@
 @pytest.fixture()
 def option_parser():
     """Returns option parser that can be used for tests."""
+    if flake8_version_info > (5, ):
+        pytest.skip('outdated test')
```
Comment (sathieu) [inline tests/test_options/conftest.py:13]: See https://github.com/wemake-services/wemake-python-styleguide/actions/runs/3153708261/jobs/5130486069

**Comment by sathieu** [comment]
@sobolevn The CI is now passing. Please review again :pray:. This was harder than expected ...

File: tests/test_options/conftest.py [hunk 3]
```diff
@@ -8,6 +9,8 @@
 @pytest.fixture()
 def option_parser():
     """Returns option parser that can be used for tests."""
+    if flake8_version_info > (5, ):
```
Comment (sobolevn) [inline tests/test_options/conftest.py:12]: I think it should be:

```python
if flake8_version_info > (5, ):
    parser = OptionManager(version=version.pkg_version, plugin_versions='', parents=[])
else:
    ...
```

Comment (sathieu) [reply tests/test_options/conftest.py:12]: I tried that, unfortunately it doesn't work:

```python
_________________________________________________________________________ test_parsing_i_dont_control_code __________________________________________________________________________

option_parser = <flake8.options.manager.OptionManager object at 0x7f88c5880e50>

    def test_parsing_i_dont_control_code(option_parser):
        """Ensures that ``i_dont_control_code`` can be parsed."""
>       args, _ = option_parser.parse_args(['--i-dont-control-code'])
E       TypeError: cannot unpack non-iterable Namespace object

tests/test_options/test_option_values/test_i_control_code.py:21: TypeError
```

I couldn't get `Application` working either.

NB: see https://github.com/PyCQA/flake8/commit/f98d52a398cd2ff5cad270fdee9e37b62444550a for the relevant upstream commit.

Comment (sathieu) [reply tests/test_options/conftest.py:12]: @sobolevn I found a solution. Please take a look :eyes: !

**Review by sobolevn — APPROVED** [review]
Looks great, thank you!

**Comment by sathieu** [comment]
Thanks @sobolevn! I'm looking forwad for a new release :wink:.

**Pull request merged**